demo/api.html

```diff
@@ -24,26 +225,51 @@ <h1 id="basics" class="page-header">
         </p>
         <p>
           The full list of events in call priority:
-          <span class="label label-warning">any</span> [
+          <span class="label label-warning">update</span> [
           <span class="label label-danger">leave</span> >
           <span class="label label-success">match</span> (
           <span class="label label-primary">enter</span> >
           <span class="label label-info">change</span> ) ].
         </p>
-        <h2>enter</h2>
+        <h3 id="rule-event-enter">
+          enter
+          <a href="rule-event-enter" class="anchorjs-link"><span class="anchorjs-icon"></span></a>
+        </h3>
         <p>
           Called when you match a rule once it wasn't triggered before.
         </p>
-        <h2>change</h2>
+        <p>
+          This event is good to append elements on DOM, children rules can rely that parent enter are executed in order
+          before children event is triggered, this way you can compose your elements in the DOM knowing that they're
+          ready to be used.
+        </p>
+        <pre class="prettyprint lang-js"><code>myRuleELement.addEventListener('enter', function () {
+  console.log('EVENT: enter from: ', this.id);
+});
+
+// Example with jQuery:
+$(myRuleELement).on('enter', function () {
+  console.log('EVENT: enter from: ', this.id);
+});</code></pre>
```

Comment:
again, a practical example of how this event is useful would be better than simply showing the boilerplate. We know the even works (or at least we have the expectation of it working, since it is being documented). What would be useful is a real-world problem that is solved with this feature